java-checks/src/main/java/org/sonar/java/checks/WeakSSLContextCheck.java

```diff
@@ -86,7 +103,18 @@ public void visitNode(Tree tree) {
           .skip(1)
           .map(secondary -> new JavaFileScannerContext.Location("Other weak protocol.", secondary))
           .toList();
-        reportIssue(unsecureVersions.get(0), "Change this code to use a stronger protocol.", secondaries, null);
+        reportIssue(unsecureVersions.get(0), ISSUE_MESSAGE, secondaries, null);
+      }
+    } else if (OPTIONS_ENABLED_PROTOCOLS.matches(mit)) {
+      ExpressionTree argument = arguments.get(0);
+      if (argument instanceof MethodInvocationTree methodInvocation && SET_OF.matches(methodInvocation)) {
+        boolean anyMatch = methodInvocation.arguments().stream()
+          .filter(arg -> arg.is(Tree.Kind.STRING_LITERAL))
+          .map(LiteralTree.class::cast)
+          .anyMatch(arg -> LiteralUtils.containsValue(arg, WEAK_FOR_SET_ENABLED_PROTOCOLS));
+        if (anyMatch) {
+          reportIssue(methodInvocation.arguments(), ISSUE_MESSAGE);
+        }
```

Comment:
I think it would be even nicer to highlight the protocols that are actually weak. Otherwise, if I have e.g. `Set.of("TLSv1.1", "TLSv1.2")`, it will highlight the entire set, even though it contains some secure protocols. An option could be to highlight the `setEnabledProtocols` method name and add all insecure protocols as secondary locations.